insights/core/__init__.py

```diff
@@ -426,6 +427,120 @@ def _scan(self):
 
         cls.scan(result_key, _scan)
 
+    def _get_after(self, timestamp, time_format='%Y-%m-%d %H:%M:%S', lines=None):
+        """
+        Find all the (available) logs that are after the given time stamp.
+
+        This method should be used by the subclass to provide an actual
+        ``get_after()`` method which provides the standard time stamp format
+        used by its particular log.  This method should only be used if the
+        subclass does not present its own or the caller needs to override
+        the default time format.
+
+        If 'lines' is not supplied, then all lines are used.  Otherwise, the
+        caller can provide a list of lines from a scanner or the ``get()``
+        method.
+
+        This method then finds all lines in that list which have a time stamp
+        after the given timestamp.  Lines that do not contain a time stamp
+        are considered to be part of the previous line and are therefore
+        included if the last log line was included or excluded otherwise.
+
+        This works by converting the time format into a regular expression
+        which matches the time format in the string.  This is then searched
+        for in each line in turn.  Only lines with a time stamp matching this
+        expression will trigger the decision to include or exclude lines.
+        Therefore, if the log for some reason does not contain a time stamp
+        that matches this format, no lines will be returned.
+
+        Note: fields that are absent will be filled in according to the rules
+        described in ``strptime()``.  For example, if the year is missing,
+        the year '1900' will be assumed.  This may cause complicated
+        behaviour in situations where the log lines cross an important
+        boundary - e.g. logs going from December to January that do not
+        include a year being searched for any line after 1st January will
+        show the lines from December.  Likewise, if your logs do not include a
+        year, make sure that you supply the appropriate year in your timestamp
+        or you will not see any lines.  This paragraph is sponsored by syslog.
+
+        No attempt is made to order the lines by date.  This also means that
+        lines can be out of order, as each timestamp is evaluated individually.
+
+        Params:
+            timestamp(datetime.datetime): lines before this time are ignored.
+            time_format(str): the ``strptime()`` string of the format of the
+                dates in this log.
+            lines(list): an optional list of lines from this parser to search.
+
+        Returns:
+            (list): a list of lines with timestamps after this date.
+        """
```

Comment:
You should add a `Raises:` section to describe how this method raises a `ParseException`.